torchao/csrc/cuda/fp6_llm/fp6_linear.cu

```diff
@@ -141,17 +149,31 @@ cudaError_t fpx_linear_kernel(cudaStream_t    stream,
         // Reduction for SplitK
         dim3 GridDim((M_Global * N_Global) / REDUCTION_ELEMENT_PER_THREADBLOCK, 1, 1);
         dim3 BlockDim(WARP_SIZE, 1, 1);
-        SplitK_Reduction<<<GridDim, BlockDim, 0, stream>>>(C, Reduction_Workspace, M_Global, N_Global, Split_K);
+        SplitK_Reduction<InputDataType><<<GridDim, BlockDim, 0, stream>>>(C, Reduction_Workspace, M_Global, N_Global, Split_K);
     }
 
     return cudaGetLastError();
```

Comment:
Either remove this (so PyTorch will check CUDA error in subsequent kernel calls), or check this error (instead of just return). If we call the function but don't check the error, the error will be reset (subsequent `cudaGetLastError()` will return success).

I think we can remove this line and make `fpx_linear_kernel()` returns `void` instead. And if you want to check the error, add `cudaGetLastError()` and necessary checks at the end of `fp_eXmY_linear_forward_cuda()`. Thoughts?

Reply:
Good catch! I missed that subsequent kernel call. I've added an error check right after the call to `SplitK_Reduction`.

The return value from `fpx_linear_kernel` wasn't used anyway, so I think it's fine to make it a `void` return function. Instead, I'm not using calls to `TORCH_CHECK` to handle failure cases inside `fpx_linear_kernel`.

Reply:
I think we still need to call `CHECK_CUDA_KERNEL_LAUNCH()` when splitK=1?

Reply:
What do you mean? The `SplitK_Reduction` kernel is only called when `splitK != 1`, so only then a call to `CHECK_CUDA_KERNEL_LAUNCH()` is required?